pkg/s3store/s3store.go

```diff
@@ -272,6 +277,70 @@ func (upload *s3Upload) writeInfo(ctx context.Context, info handler.FileInfo) er
 	return err
 }
 
+type s3ChunkProducer struct {
```

Comment:
Can you add a brief sentence here documenting what this structure does? Also, what about naming it s3PartProducer to align with the naming of MaxBufferedParts?